cohorts/load.py

```diff
@@ -652,16 +697,26 @@ def load_effects(self, patients=None, only_nonsynonymous=False, variant_type="sn
         patient_effects = {}
         for patient in self.iter_patients(patients):
             effects = self._load_single_patient_effects(
-                patient, only_nonsynonymous, variant_type, merge_type, filter_fn)
+                patient, only_nonsynonymous, variant_type, merge_type, only_expressed, filter_fn)
             if effects is not None:
                 patient_effects[patient.id] = effects
         return patient_effects
 
     def _load_single_patient_effects(self, patient, only_nonsynonymous,
-                                     variant_type, merge_type, filter_fn):
+                                     variant_type, merge_type,
+                                     only_expressed, filter_fn):
+        def on_return(variants, effects):
+            if only_expressed:
+                variants = self._filter_variants_to_expressed(variants,
+                                                              patient=patient,
+                                                              variant_type=variant_type,
+                                                              merge_type=merge_type)
+                effects = [effect for effect in effects if effect.variant in variants]
+            return filter_effects_with_metadata(effects, variants.metadata, filter_fn)
```

Comment:
This works, but the first arg of `filter_effects_with_metadata` was supposed to be an `EffectCollection`